runtime/onert/backend/gpu_cl/ClMemoryManager.h

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (c) 2021 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef __ONERT_BACKEND_ACL_COMMON_MEMORY_MANAGER_H__
+#define __ONERT_BACKEND_ACL_COMMON_MEMORY_MANAGER_H__
+
+#include <cassert>
+
+#include "ir/OperandIndexMap.h"
+#include "../open_cl/InferenceContext.h"
+#include "../open_cl/StorageTypeUtil.h"
+#include "../open_cl/ClContext.h"
+#include "../open_cl/TensorType.h"
+#include "util/logging.h"
+#include "ir/Shape.h"
+
+namespace onert
+{
+namespace backend
+{
+namespace gpu_cl
+{
+
+template <typename T_ITensor, typename T_Tensor> class ClMemoryManager
+{
+public:
+  ClMemoryManager(CLContext *context) : _context{context} {}
+
+  virtual ~ClMemoryManager() = default;
+
+  virtual void allocate(void)
+  {
+    for (const auto &tensor_entry : _tensors)
+    {
+      auto tensor = tensor_entry.second;
+      const auto &t = tensor_reserver_.Get(tensor_entry.first.value());
+      const auto &shape = t.shape;
+      const auto &descriptor = t.descriptor;
+      CreateTensor(*_context, shape, descriptor, tensor->handle());
+    }
+  }
+
+  virtual void deallocate(void)
+  {
+    // for (const auto &tensor_entry : _tensors)
```

Comment:
If those lines should be updated, Would you like replace with "ToDo - ***"?

Reply:
okay